jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -2068,6 +2070,35 @@ private ConfigurationParser(Pool.Entry<ConfigurationParser> entry)
             addCatalog(URI.create(catalogUrl.toExternalForm()));
         }
 
+        protected SAXParserFactory newSAXParserFactory()
+        {
+            // Use JVM default implementation (not the one found in the classloader from non-JVM sources)
+            SAXParserFactory factory = SAXParserFactory.newDefaultInstance();
+            // Don't allow use of XInclude to reference external entities.
+            factory.setXIncludeAware(false);
+            return factory;
+        }
+
+        @Override
+        public void setValidating(boolean validating)
+        {
+            super.setValidating(validating);
+
+            try
+            {
+                XMLReader xmlReader = getSAXParser().getXMLReader();
+
+                // disable all external entity references with Jetty's Configuration XML.
+                xmlReader.setFeature("http://xml.org/sax/features/external-general-entities", false);
+                xmlReader.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+                xmlReader.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+            }
+            catch (SAXException e)
+            {
+                LOG.warn(e.getMessage());
+            }
```

Comment:
This feel a bit like a side-effect that known the lifecycle of the sax parsers.  Could this not be done in a wrapper of the `SAXParserFactory` created by `newSAXParserFactor()`?

Reply:
This is pretty common, we do some of this in the parent `XmlPparser`, but not all of it, but not all of it.
It's not appropriate to have the generic `XmlParser` be this strict, as it's used in other places (like web descriptor parsing).
Only the `XmlConfiguration` needs this level of strictness.

Reply:
I'm only proposing that we do it for the SAXParserFactory that is returned above for the `XmlConfiguration`

Reply:
Changing the "generic" XML behavior in `XmlParser` doesn't seem right, as it's used in many scenarios where an alternative XML parser would make sense, and even accessing external entities could make sense.

The only scenario where we want to be strict is `XmlConfiguration`, hence why it is there.

Reply:
@joakime I'm only proposing this for XMLConfiguration. But do it in a wrapper in the newSAXParserFactory method rather than as a side effect of a setter based on assumed knowledge of the parser lifecycle